source/common/router/config_impl.cc

```diff
@@ -664,15 +666,23 @@ std::string RouteEntryImplBase::newPath(const Http::RequestHeaderMap& headers) c
     final_host = processRequestHost(headers, final_scheme, final_port);
   }
 
+  current_path = headers.getPathValue();
```

Comment:
I think this would be a bit easier to follow if `current_path` was just declared and used within the `!path_redirect_.empty()` block.

Reply:
if `path_redirect_` is empty, we would need to set `final_path` to `current_path`.